### PR TITLE
Update point clouds to default to `z` instead of first field

### PR DESCRIPTION
**Pull request by snosenzo**
**User-Facing Changes**

- Point clouds will default to color by `z` when intensity and rgb fields aren't present

**Description**

Updated to color by `z` as a fallback for all colorMode applicable renderables. Also in this I consolidated the autoselecting of color settings, and updated all renderables to use the same method. I also consolidated the setting of fields for pointlcouds since `isSupported` would run on both foxglove and ros pointclouds, and we should be using it as a flag to determine whether to update the autoselection of color settings aside from the colorField being `undefined` because the colorField can be undefined for `rgba-fields`.

We're also defaulting to `x` in `colorMode.ts` [here](https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/ThreeDeeRender/renderables/colorMode.ts#L199), which we use for FoxgloveGrid, and Images. ~~Should I update that to use `z` as well?~~ updated. 

Fixes: FG-5831

### CODE REVIEW

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/pointExtensionUtils.ts [hunk 1]
```diff
@@ -179,10 +174,18 @@ export function autoSelectColorField(
     }
   }
 
+  // Next check to see if there is a `z` field
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/pointExtensionUtils.ts:177]:
```suggestion
  // `z` field is third priority
```

**Review by jtbandes — APPROVED** [review]
Yeah I would probably change the other one too for consistency. I doubt most people are actually using it though, because `x` is a weird field name to have in a grid... btw is there a good reason that the other code isn't using this autoSelectColorField function?

**Comment by snosenzo** [comment]
@jtbandes re-requesting review because I consolidated the autoselecting of colorsettings and updated a number of places

**Review by jtbandes — APPROVED** [review]
Nice, thanks for the cleanup

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/FoxgloveGrid.ts [hunk 2]
```diff
@@ -493,6 +492,15 @@ export class FoxgloveGrid extends SceneExtension<FoxgloveGridRenderable> {
       return;
     }
 
+    let fields = this.#fieldsByTopic.get(topic);
+    let fieldsUpdated = false;
+    if (!fields || fields.length !== foxgloveGrid.fields.length) {
```
[the remark this answers is not in the document]
Comment (snosenzo) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/FoxgloveGrid.ts:497]: I agree, we also do this for point clouds. haven't heard any complaints so far though 🤷

**Pull request merged**